microsoft/PowerToys
# [FancyZones] Configurable sensitivity radius

## Summary of the Pull Request
In the current version the FancyZones use a const of 20px when determining if two zones should be highlighted/window should be placed in two zones. These 20px are not very much on high resolution displays. This Pull request changes the const to a configurable setting per layout.

## PR Checklist
* [x] Applies to #6437 
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #6437 

## Info on Pull Request
- Add Setting the in FancyZone Editor for setting the Sensitivity radius
- Update XAML in editor to de-duplicate some code 
- Show settings (also the spacing setting) for both default and custom layouts
- Set default of 20px when setting is not yet present in json (i.e. upgrade scenario)

## Validation Steps Performed

Upgrade scenario:
- Have FancyZones configured
- Start new version
- JSON settings adds default 20 px for "editor-sensitivity-radius"

Usage:
- Open the FancyZone editor 
- New setting available "Space inside zone to highlight adjacent zone"
- Set the value to some higher value
- Apply setting
- Drag a window, using FancyZones, in between zones
- Both zones light up
- Now move away from the zone's edge. and see that the other zone now remains hightlighted for the ammount of pixels you set
